### DISCUSSION TITLE
pdf and image search capabilities?

### DISCUSSION THREAD

**Discussion by fkgruber** [khoj-ai/khoj#344] [Q&A]
What are the capabilities of the pdf and image search? 
recognize objects? handwriting? print text?, recognize math?

**Comment by sabaimran** [comment]
Hey @fkgruber! PDF indexing will take in any of the text in the PDF, but it won't do additional processing on any images within the document. Images works well for object, place, action queries. For example "Hiking Mt. Rainier", "Falcons flying by a tree", "classroom at night" should all be good questions, but "x+y=5" or searching for handwriting (like OCR) won't work as well.

Do you have related feature requests or use cases in mind?

I think we answered this [in Discord as well](https://discord.com/channels/1112065956647284756/1112065956647284759/1131974619386089487), but duplicating in case anyone else has this question in Github.

**Comment by fkgruber** [comment]
Hi
I tried the pdf search and while I get results I don't see any link to the page or location in the pdf where this notes were found. For example,

<img width="1295" alt="image" src="https://github.com/khoj-ai/khoj/assets/4311763/5488c499-4b4f-40dd-b777-9a41fb14f21a">

Is this fixed in the latest version or development version? I'm still on 0.9.0

**Reply by sabaimran** [reply]
Ah, we can't quite support that in PDFs at the moment. Maybe you know more, but I don't know of a way to deep link into a PDF file.

**Reply by fkgruber** [reply]
you can easily link to a specific page number in the pdf:
[[docview:~/mypdf.pdf::3][link to page 3]]

or if you use pdf-tools and org-pdftools 
[[pdf:~/Documents/gitRepos/general-functions/org/causal.pdf::3][link to page 3]]

you can also link to specific locations in your pdf. I generated the following link with org-store-link after selecting some text in the pdf and then org-insert-link on the org document.
[[pdf:~/Documents/gitRepos/general-functions/org/causal.pdf::5++11.62;;annot-5-0][causal.pdf: Page 5; Quoting: Measures of Interaction for binary treatment and expo- sure]]

If I click on the link it opens the pdf in the specific location at specific zoom level. I think it creates some sort of invisible annotation

see
https://github.com/fuxialexander/org-pdftools

**Comment by fkgruber** [comment]
I just tried with 0.10.1 and I'm seeing the same behavior? Do I need to reconfigure or reset?
